### ISSUE TITLE
Contributing code - and assigning usage rights in perpetuity

### ISSUE THREAD

**Issue by binarytemple** [TI-Tokyo/riak-community-admin#11]
In terms of contributing code to the project there comes the small question of rights - suggest we use something like this - https://github.com/gchq/Gaffer/wiki/GCHQ-OSS-Contributor-License-Agreement-V1.0 - it's got a nice little email template link at the bottom of the text - suggest the email goes to a google groups Riak group.

**Comment by pjaclark** [comment]
Results from 2017-09-15:

Apache2 license includes a CLA that is easily understandable.
A separate CLA would increase the paperwork and deter contributors who are already covered by the Apache2 license.

More comments/feedback appreciated, and will revisit next week.

**Comment by jadeallenx** [comment]
Strongly 👎 on this proposal. Please please please please please do not use a separate CLA.

**Comment by pjaclark** [comment]
Anyone with a company legal team who can give an opinion on the need for a CLA?

**Comment by jadeallenx** [comment]
Basho never used one. I have no idea why one is needed now.

**Comment by binarytemple** [comment]
Nobody ever contributed to Riak except sister companies like ESL and Trifork ? 

Opencredo have some illuminating metrics (opportunistic, but truthful) 

https://opencredo.com/riak-the-dynamo-paper-and-life-beyond-basho/

**Comment by jadeallenx** [comment]
Sort of proves my point: the number of contributions outside of a few people is limited already. Why limit the pool further with a CLA?

**Comment by binarytemple** [comment]
Just to be clear, I’m not advocating signing a PDF, faxing a signed document or suchlike busy working, paper pushing… 

I’m advocating that they send a message to a public google groups mailing list that they already participate in saying: 

“<date> I’m making some contributions to the project - they will be under the Apache 2 CLA - my (real) name is <joe blogs> - have a nice day. "

And that statement sits there in perpetuity. Nothing more. I’d do it. 

Unless... I was working for a company that made me sign an employment contract with a clause that said, “we own any creative output, invention or patent you file, for the duration of your employment, unless specific exemption is granted in writing”.

And that clause is very common.

**Comment by jadeallenx** [comment]
I've said all I can constructively say about CLAs. I will not be persuaded they do anything helpful. I will have to agree to disagree.

**Comment by russelldb** [comment]
I agree with Mark Allen. Basho _did_ have contributions from non-"sister" companies (NHS's 2i work for example, or DVVSet from Tomé) without need of a CLA.

**Comment by Licenser** [comment]
So here is my reasoning for **NOT** having an extra CLA, there are multiple points to take into account.

1) It does not do any good, as @mrallen1 already pointed out there are outside code contributions in riak and it's libraries not to mention the outside libraries it uses. So we would not have a 'clean slate' anyway. Would we rip out prior contributions of people that didn't sign a CLA? Would we just ignore that fact?

2) I never heard of any software being chosen (or discarded) because they have or have not a CLA. If that were the case Linux would not be used, or Docker or Riak as a matter of fact.

3) A CLA scares away new contributors, and this is not a hypothetical this is a fact, I personally have decided against contributing to projects because of CLAs and there are certainly others who have done the same.

4) A affects new contributors more than large contributors, someone who likes riak and just wants to do a small contribution, in the beginning, will quite possibly just walk away if they have to spend more time reading and understanding a CLA then they have to spend actually contributing. Many people start off with contributing this way and when they find a welcoming community stay, we would kill that off in the beginning.

5) Sorry to put it so harshly but it is discriminatory, and with that directly against what we claim in the CoC we want to adopt `The Riak community is made up of members from around the globe with a diverse set of skills, personalities, and experiences.` (#7). Unless we expect people to sign a contract (or agreement) without understanding what it means a CLA discriminates:
a) against non-UK citizens: as they do not know UK law.
b) against non-native speakers: as they will struggle with understanding legal wording.
c) against people who have enjoyed less education: as legal terms are quite honestly complicated.
d) against people with little means: as they can not consult a lawyer on the matter.

6) It causes real harm to prevent theoretical harm. The only reason for a CLA is the maybe-case that someone acts in bad faith, which is sad and horrible if it happens but for once the APL already protects against most of that, and second, this is a case that very likely **will never occure**. On the other hand, the impact of a CLA is real, direct and measurable from the second it is implemented.

7) It puts the burden on those with little means. If a large company is concerned about the licensing state of a project they have the means to ask their legal team for advice and possibly solve their problem. Individual contributors, who give their free time and energy just to help do not. A CLA shifts the burden for companies to individual contributors.

In short: I find a CLA a completely unacceptable solution.

**Comment by Licenser** [comment]
Oh, one more thing since https://github.com/gchq/Gaffer is the example here I looked at the repo and it's worth noting that they've virtually zero outside contributors. Of cause, correlation does not imply causation, but it's still something interesting to observe.

**Comment by pjaclark** [comment]
From 2017-09-22:

Switching from a CLA (as in Apache's ones) to a DCO (Developer Certificate of Origin) (as in Linux's one).

Sample is here: http://elinux.org/Developer_Certificate_Of_Origin

Idea is that it is shorter, simpler, and covers they key items needed to cover us legally.

Suggestions on exact wording and usage appreciated for next week.

**Comment by Licenser** [comment]
I'm still unsure if a majority of people even want any of this, based on the comments here it's a bit on the edge one way or another.

**Comment by jadeallenx** [comment]
DCO is the *least* objectionable option if there in fact **MUST** be one. P.S. Still not in favor of either option.

**Comment by sata** [comment]
I wouldn't have a problem with the wording in that DCO. Sounds like a perfect solution. I can't see how this DCO would push any contributor away.

**Comment by pjaclark** [comment]
Decisions from 2017-09-29:

- Propose to use the Linux one as a model (http://elinux.org/Developer_Certificate_Of_Origin)
- Make this one of the first things for the Project Direction Group to review.
